macros/dedupe_artifacts_v2.sql

```diff
@@ -0,0 +1,34 @@
+{% macro dedupe_dbt_artifacts_v2() %}
+
+{% set src_results = source('dbt_artifacts', 'dbt_run_results') %}
+{% set src_results_nodes = source('dbt_artifacts', 'dbt_run_results_nodes') %}
+{% set src_manifest_nodes = source('dbt_artifacts', 'dbt_manifest_nodes') %}
+
+{% for artifact_table in [src_results, src_results_nodes, src_manifest_nodes] %}
+
+    {% set dedupe_results_query %}
+
+        create temporary table {{ artifact_table.database }}.{{ artifact_table.schema }}.dbt_temp_artifact_table as
+            select * from {{ artifact_table }}
+            qualify row_number() over (

```

Comment:
```suggestion
            qualify rank() over (

```

Thinking about it, should this be a rank? The `src_results_nodes` and `src_manifest_nodes` will and should have multiple entries for a single `command_invocation_id`.

Reply:
Hmmmmmm - you're totally right on the issue, but I think wrong on the solution. Will take a look a revamp this.

Reply:
Ah yeah rank wouldn't work. Shouldn't it be as simple as:
```
qualify row_number() over (
    partition by command_invocation_id, node_id
    order by artifact_generated_at asc
) = 1;
```

actually?

Reply:
read my mind :)

See my most recent commit.

Reply:
Love the table_key lookup!